nengo/spa/vocab.py

```diff
@@ -147,6 +149,17 @@ def add(self, key, p):
 
         The pointer value can be a SemanticPointer or a vector.
         """
+        if self.read_only:
+            if self.parent is None:
+                # If this vocabulary is not a subset vocabulary, throw an error
+                raise ValueError("Cannot add semantic pointer '%s' to "
+                                 "read-only vocabulary." % key)
+            else:
+                # If this vocabulary is a subset vocabulary, apply operation on
+                # parent vocab.
+                self.parent.add(key, p)
+                return
```

Comment:
This is to handle the cases where something is added to a vocabulary that is a subset of another vocabulary. Basically because of the magic that happens in the `transform_to` function (it add keys from one vocabulary to another) of the `Vocabulary` class. 

If we made it a `ValueError` this would fail:
```
vocab1 = Vocabulary(32)
vocab1.parse('A+B+C')
vocab2 = vocab1.create_subset(['A', 'B'])

with spa.SPA() as model:
    model.state_in = spa.State(vocab=vocab1)
    model.am = spa.AssociativeMemory(vocab2)
    model.state_out = spa.State(vocab=vocab1)

    cortical_actions = spa.Actions('am=state_in', 'state_out=am')
    
    model.cortical = spa.Cortial(cortical_actions)
```

The transform_to system will try to add `'C'` to `vocab2` and fail with the `ValueError`.